library/std/src/sys/pal/unix/process/process_common.rs

```diff
@@ -580,8 +580,29 @@ impl fmt::Debug for Command {
 
             debug_command.finish()
         } else {
+            /// Only escape arguments when necessary, otherwise output it
+            /// unescaped to make the output easier to read.
+            fn relaxed_escaping(bytes: &[u8]) -> impl fmt::Display + use<'_> {
+                // Don't escape if all the characters are likely
+                // to not be a problem when copied to the shell.
+                fmt::from_fn(move |f| {
+                    if !bytes.is_empty()
+                        && bytes.iter().all(|c| {
+                            c.is_ascii_alphanumeric()
+                                || matches!(c, b'_' | b'/' | b'-' | b'=' | b'.' | b',')
+                        })
```

Comment:
Note that I don't know the first thing about shells and their escaping rules, so this might be completely wrong

Reply:
https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_02

Reply:
Fantastic, thanks!